spotbugs/src/test/java/edu/umd/cs/findbugs/util/ClassNameTest.java

```diff
@@ -65,4 +65,10 @@ public void testGetPrimitiveType() {
     public void testExtractClassNameBad() {
         ClassName.extractClassName("L[Ljava/lang/Integer;");
     }
+    
+    @Test
+    public void testAssertIsSlashedTrue() {
```

Comment:
Pardon me if I am wrong.
My current understanding is these are just developer checks(verifying assumptions) which could also be turned off while running it. If the assumption fails these asserts will start to fail but could be turned off to bypass them.

As assertIsSlashed is not used, so no harm in removing it, just I feel it is been there just to preserve symmetry with assertIsDotted.

I am not totally sure on removing these methods; but I would like to make some code change with my current understanding. Will update this review with same.

Reply:
I have never used assert keyword yet, so this is a new experience for me.
After reading a bit I got an understanding of how it could be helping the developer in verifying his assumption which is based on something 3rd party behavior, which is not in his control.
Like [this code](https://github.com/spotbugs/spotbugs/blob/release-3.1/spotbugs/src/main/java/edu/umd/cs/findbugs/AbstractBugReporter.java#L221)

Reply:
I only worked by using exceptions to prevent such behavior, so I am also okay with removing these methods.
Just let me know what would be the right path to proceed. :-)